### PR TITLE
Add UWP display prompt

### LINKED ISSUE #6713: [Enhancement] Display prompts
## Summary
Forms has the ability to show action sheets and alerts but no prompts. I tried using ACR User Dialogs but found out that the Android implementation is broken (soft keyboard does not appear upon initial display, incorrect keyboard type, no margins on the text input, etc.) This is not a 1-to-1 port of ACR prompts but a simple implementation that should work in most cases.

## API Changes

```
Task<string> Page.DisplayPromptAsync([arguments])
PromptArguments => will hold config data
```

## Intended Use Case
Any time when the user is asked to input value before proceeding to do what they want. User action is required in a modal view because it's important.

### PR DESCRIPTION
**Pull request by bmacombe**
### Description of Change ###

Adds UWP Display Prompt

### Issues Resolved ### 

- fixes for UWP #6713 - 

### API Changes ### 
 None

### Platforms Affected ### 

- UWP

### Behavioral/Visual Changes ###
Adds Display Prompt

### Before/After Screenshots ### 

image

image

image

### Testing Procedure ###
Use Issue 6713 with UWP

### PR Checklist ###

- [ ] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

**Comment by bmacombe** [comment]
Not ready for review, I'll take it out of draft when ready

**Comment by bmacombe** [comment]
Ready for review.  I was only able to compare functionally against android as I don't have a mac available at the moment to build the iOS control gallery

**Comment by jfversluis** [comment]
Maybe I'm overlooking it, but did you include the `InitialValue` as well? See #8362

File: Xamarin.Forms.Platform.UAP/Platform.cs [hunk 1]
```diff
@@ -530,6 +533,45 @@ static void OnPageActionSheet(object sender, ActionSheetArguments options)
 			}
 		}
 
+		static async void OnPagePrompt(Page sender, PromptArguments options)
+		{
+			var promptDialog = new PromptDialog
+			{
+				Title = options.Title ?? string.Empty,
+				Message = options.Message ?? string.Empty,
+				Input = options.InitialValue ?? string.Empty,
```
Comment (bmacombe) [inline Xamarin.Forms.Platform.UAP/Platform.cs]: @jfversluis Yep, should be right here.  It's also shown in the third screen shot

Comment (jfversluis) [reply Xamarin.Forms.Platform.UAP/Platform.cs]: 🤦‍♂️ great stuff! Thanks!

File: Xamarin.Forms.Platform.UAP/Xamarin.Forms.Platform.UAP.csproj [hunk 2]
```diff
@@ -256,6 +259,10 @@
       <SubType>Designer</SubType>
       <Generator>MSBuild:Compile</Generator>
     </Page>
+    <Page Include="PromptDialog.xaml">
```
Comment (samhouts) [inline Xamarin.Forms.Platform.UAP/Xamarin.Forms.Platform.UAP.csproj]: This won't be packaged unless it's added to the nuspec.

Comment (bmacombe) [reply Xamarin.Forms.Platform.UAP/Xamarin.Forms.Platform.UAP.csproj]: Thanks...I didn't realize that I'll take a look.

File: .nuspec/Xamarin.Forms.nuspec [hunk 3]
```diff
@@ -229,6 +229,7 @@
     <file src="..\Xamarin.Forms.Platform.UAP\bin\$Configuration$\Shell\ShellStyles.xbf" target="lib\uap10.0\Xamarin.Forms.Platform.UAP\Shell" />
     <file src="..\Xamarin.Forms.Platform.UAP\bin\$Configuration$\PickerStyle.xbf" target="lib\uap10.0\Xamarin.Forms.Platform.UAP" />
     <file src="..\Xamarin.Forms.Platform.UAP\bin\$Configuration$\Microsoft.UI.Xaml\Themes\generic.xbf" target="lib\uap10.0\Xamarin.Forms.Platform.UAP\Microsoft.UI.Xaml\Themes\" />
+    <file src="..\Xamarin.Forms.Platform.UAP\bin\$Configuration$\PromptDialog.xbf" target="lib\uap10.0\Xamarin.Forms.Platform.UAP" />
```
Comment (bmacombe) [inline .nuspec/Xamarin.Forms.nuspec]: @samhouts I believe I updated the nuspec...to be honest,  I have no idea what I'm doing there...but I think I made a good guess based on other files from the UWP platform...would you please double check?  Thanks!

Comment (PureWeen) [reply .nuspec/Xamarin.Forms.nuspec]: @bmacombe can you rebase this PR? The latest master doesn't require any of these to be specified on the nuspec

Comment (bmacombe) [reply .nuspec/Xamarin.Forms.nuspec]: @PureWeen Sure, I'll try to do that in the next few days

Comment (bmacombe) [reply .nuspec/Xamarin.Forms.nuspec]: @PureWeen After a rebase I'm getting this error when trying to compile

error MSB4184: The expression "[System.Version]::Parse('')" cannot be evaluated. Version string portion was too short or too long.

Any thoughts?

Comment (PureWeen) [reply .nuspec/Xamarin.Forms.nuspec]: delete your obj/bin directories and then you should be fine

Comment (bmacombe) [reply .nuspec/Xamarin.Forms.nuspec]: @PureWeen I did that a couple times and I'm down to this error on XF.Core.

Severity	Code	Description	Project	File	Line	Suppression State
Error	MSB4062	The "XFCorePostProcessor.Tasks.FixXFCoreAssembly" task could not be loaded from the assembly C:\git\Xamarin.Forms\Xamarin.Forms.Core\..\XFCorePostProcessor.Tasks\bin\Debug\net461\XFCorePostProcessor.Tasks.dll. Could not load file or assembly 'file:///C:\git\Xamarin.Forms\XFCorePostProcessor.Tasks\bin\Debug\net461\XFCorePostProcessor.Tasks.dll' or one of its dependencies. The system cannot find the file specified. Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask.	Xamarin.Forms.Core	C:\git\Xamarin.Forms\Xamarin.Forms.Core\Xamarin.Forms.Core.csproj	33

Comment (PureWeen) [reply .nuspec/Xamarin.Forms.nuspec]: your best bet will probably be to cleanout the directory with *git clean -xdf* 

keep in mind that anything you haven't checked in will get deleted

Comment (bmacombe) [reply .nuspec/Xamarin.Forms.nuspec]: @PureWeen That seemed to do the trick...thanks for that tip, I didn't know that one.  Rebase pushing shortly.

Comment (PureWeen) [reply .nuspec/Xamarin.Forms.nuspec]: Yay! We've been going through some pretty heavy csproj changes which unfortunately don't "just work" when they get merged in

**Pull request merged**